Sources/PostgresNIO/Connection/PostgresConnection+Configuration.swift

```diff
@@ -192,9 +192,22 @@ extension PostgresConnection {
         /// - Parameters:
         ///   - channel: The `NIOCore/Channel` to use. The channel must already be active and connected to an
         ///     endpoint (i.e. `NIOCore/Channel/isActive` must be `true`).
-        ///   - tls: The TLS mode to use. Defaults to ``TLS-swift.struct/disable``.
+        ///   - tls: The TLS mode to use.
+        public init(establishedChannel channel: Channel, tls: PostgresConnection.Configuration.TLS, username: String, password: String?, database: String?) {
+            self.init(endpointInfo: .configureChannel(channel), tls: tls, username: username, password: password, database: database)
+        }
+        
+        /// Create a configuration for establishing a connection to a Postgres server over a preestablished
+        /// `NIOCore/Channel`.
+        ///
+        /// This is provided for calling code which wants to manage the underlying connection transport on its
+        /// own, such as when tunneling a connection through SSH.
+        ///
+        /// - Parameters:
+        ///   - channel: The `NIOCore/Channel` to use. The channel must already be active and connected to an
+        ///     endpoint (i.e. `NIOCore/Channel/isActive` must be `true`).
         public init(establishedChannel channel: Channel, username: String, password: String?, database: String?) {
-            self.init(endpointInfo: .configureChannel(channel), tls: .disable, username: username, password: password, database: database)
+            self.init(establishedChannel: channel, tls: .disable, username: username, password: password, database: database)
```

Comment:
Let's revert this here and make this change in #526, which we can land after #527.

Reply:
This PR is useless without #526 as far as I understand

Reply:
This PR enables #526 to work correctly. Therefore we should land it first and then land #526. But we should not add the changes that we want to make in #526 here.

Reply:
I needed a working branch locally so I had to base one on the other branch 🙃